src/submission/submission-to-clinical.ts

```diff
@@ -5,17 +5,63 @@
 import { DeepReadonly } from 'deep-freeze';
 import { Donor, Specimen, Sample } from '../clinical/clinical-entities';
 import {
+  ActiveClinicalSubmission,
+  CommitClinicalSubmissionCommand,
   CommitRegistrationCommand,
   ActiveRegistration,
   SubmittedRegistrationRecord,
   FieldsEnum,
+  SUBMISSION_STATE,
 } from './submission-entities';
 
 import { Errors } from '../utils';
 import { donorDao } from '../clinical/donor-repo';
 import _ from 'lodash';
 import { F } from '../utils';
 import { registrationRepository } from './registration-repo';
+import { submissionRepository } from './submission-repo';
+import { mergeActiveSubmissionWithDonors } from './merge-submission';
+
+/**
+ * This method will move the current submitted clinical data to
+ * the clinical database
+ *
+ * @param command CommitClinicalSubmissionCommand with the versionId of the registration to close
+ */
+export const commitClinicalSubmission = async (
+  command: Readonly<CommitClinicalSubmissionCommand>,
+) => {
+  // Get active submission
+  const activeSubmission = await submissionRepository.findByProgramId(command.programId);
+
+  if (activeSubmission === undefined) {
+    throw new Errors.NotFound('No active submission data found for this program.');
+  } else if (activeSubmission.version !== command.versionId) {
+    throw new Errors.InvalidArgument(
+      'Version ID provided does not match the latest submission version for this program.',
+    );
+  } else if (activeSubmission.state !== SUBMISSION_STATE.VALID) {
+    // confirm that the state is VALID
+    throw new Errors.StateConflict(
+      'Active submission does not have state VALID and cannot be committed.',
+    );
+  } else {
+    // We Did It! We have a valid active submission to commit! Everyone cheers!
+    // Get all the current donor documents
+    const donorDTOs = await getDonorDTOsForActiveSubmission(activeSubmission);
```

Comment:
like we talked about could use `getDonorsInProgram()` from submission-service

Reply:
Not going to refactor into using a map, but I did use the existing donor-repo fetch method, removed the redundant implementation